services/inputhost/pathCache.go

```diff
@@ -508,3 +508,16 @@ func (pathCache *inPathCache) getState() *admin.DestinationState {
 
 	return destState
 }
+
+func (pathCache *inPathCache) drainExtent(extUUID string) error {
+	pathCache.RLock()
+	defer pathCache.RUnlock()
+
+	extCache, ok := pathCache.extentCache[extentUUID(extUUID)]
```

Comment:
Yeah i am sending the next PR which will drain the pubconnection appropriately. I need to do (3) and yes i have a guarantee to do that. 

And note, these are all best effort so if a connection is stuck for some reason (say a socket Write is blocking, I don't want to wait there forever).